src/security-testing/MockVulnerableComponent.js

```
@@ -0,0 +1,98 @@
+
+import React, { useState, useEffect } from 'react';
+
+const MockVulnerableComponent = () => {
+  const [userInput, setUserInput] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+  const [debugMode, setDebugMode] = useState(false);
+
+  // Mock vulnerability 1: XSS via dangerouslySetInnerHTML
+  // This simulates a common XSS vulnerability where user input is rendered without sanitization
+  const renderUserContent = (content) => {
+    return <div dangerouslySetInnerHTML={{ __html: content }} />;
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)